### PR TITLE
feat: prompt interaction time

### PR DESCRIPTION
**Pull request by ammarkarachi**
#### Description of changes
Added additional field that measure prompt time

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sachscode — COMMENTED** [review]
LGTM, just one question.

**Review by lazpavel — APPROVED** [review]
LGTM, added nit for linter auto added comments

File: packages/amplify-cli/src/domain/amplify-usageData/IUsageData.ts [hunk 1]
```diff
@@ -52,6 +55,8 @@ export enum ManuallyTimedCodePath {
   // init-specific paths (also called during env checkout and pull)
   INIT_ENV_PLATFORM = 'initEnvPlatform', // time to call awscloudformation provider initEnv. This includes downloading deployment bucket and updating local files
   INIT_ENV_CATEGORIES = 'initEnvCategories', // time to call all of the category's initEnv methods
+
+  PROMPT_TIME = 'promptTime'// total time to takes to answer a prompt
```
[the remark this answers is not in the document]
Comment (ammarkarachi) [reply packages/amplify-cli/src/domain/amplify-usageData/IUsageData.ts:59]: No, do you have suggestions?

File: packages/amplify-prompts/src/stopwatch.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      this._currentState = StopwatchState.IDLE;
+      this._slices.length = 0;
+    }
+
+    /**
+     * pauses the time call start to resume
+     */
+    pause = (): void => {
+      if (this._currentState === StopwatchState.IDLE) {
+        throw new Error('Cannot pause no timer running');
```
Comment (danielleadams) [inline packages/amplify-prompts/src/stopwatch.ts:29]: Will this terminate/error out the entire CLI process if the state gets out of sync? I can't really tell how it catches at the prompter level, but it seems like the stop watch should gracefully handle and persist the error on our end without stopping the developer's CLI command.

**Review by sachscode — APPROVED** [review]
LGTM

File: packages/amplify-cli/src/domain/amplify-usageData/IUsageData.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sachscode) [reply packages/amplify-cli/src/domain/amplify-usageData/IUsageData.ts:59]: No worries, we can attach the relevant labels post processing

File: packages/amplify-prompts/src/stopwatch.ts (hunk 2 above)
Comment (sachscode) [reply packages/amplify-prompts/src/stopwatch.ts:29]: +1

Comment (ammarkarachi) [reply packages/amplify-prompts/src/stopwatch.ts:29]: Sounds good let me make that change

**Pull request merged**